server/models/dataset.py

```diff
@@ -326,9 +328,124 @@ def registerMetadata(self, dataset, csvFile, user, sendMail=False):
                     'host': host,
                     'dataset': dataset,
                     'user': user,
-                    'csvFile': csvFile,
+                    'metadataFile': metadataFile,
                     'date': now.replace(microsecond=0)
                 },
                 subject='ISIC Archive: Dataset Metadata Notification')
 
         return dataset
+
+    def applyMetadata(self, dataset, metadataFile, save):
+        """
+        Apply metadata in a .csv file to a dataset.
+        :return: A list of strings describing parsing or validation errors.
+        :rtype: list
+        """
+        Assetstore = self.model('assetstore')
+
+        assetstore = Assetstore.getCurrent()
+        assetstoreAdapter = assetstore_utilities.getAssetstoreAdapter(
+            assetstore)
+        metadataFilePath = assetstoreAdapter.fullPath(metadataFile)
```

Comment:
The abandoned PR (https://github.com/ImageMarkup/isic-archive/pull/310) includes a change to support general assetstores. We could pull a version of that in at some point.